x-pack/plugins/security_solution/public/detections/components/rules/step_rule_actions/actions_view.tsx

```diff
@@ -0,0 +1,218 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+
+import React from 'react';
+import { EuiToolTip, EuiText, EuiSpacer, EuiFlexGroup, EuiFlexItem, EuiIcon } from '@elastic/eui';
+import type { RuleActionFrequency, RuleAction } from '@kbn/alerting-plugin/common';
+import type { ActionType, AsApiContract } from '@kbn/actions-plugin/common';
+import type { ActionResult } from '@kbn/actions-plugin/server';
+import type { ActionTypeRegistryContract } from '@kbn/triggers-actions-ui-plugin/public';
+import { FormattedMessage } from '@kbn/i18n-react';
+import {
+  useFetchConnectors,
+  useFetchConnectorTypes,
+} from '../../../../detection_engine/rule_management/api/hooks/use_fetch_connectors';
+import { useKibana } from '../../../../common/lib/kibana';
+import type { ActionsStepRule } from '../../../pages/detection_engine/rules/types';
+import { getActionDetails } from '../../../../detection_engine/rule_response_actions/constants';
+import { StepPanel } from '../step_panel';
+import * as i18n from './translations';
+
+const DescriptionLine = ({ children }: { children: React.ReactNode }) => (
+  <EuiFlexItem>
+    <EuiText size="xs" color="subdued">
+      {children}
+    </EuiText>
+  </EuiFlexItem>
+);
+
+export const FrequencyDescription: React.FC<{ frequency?: RuleActionFrequency }> = ({
+  frequency,
+}) => {
+  if (!frequency) {
+    return null;
+  }
+
+  if (!frequency.summary) {
+    return <DescriptionLine>{i18n.FOR_EACH_ALERT_PER_RULE_RUN}</DescriptionLine>;
+  }
+
+  if (frequency.notifyWhen === 'onActiveAlert') {
+    return <DescriptionLine>{i18n.SUMMARY_OF_ALERTS_PER_RULE_RUN}</DescriptionLine>;
+  }
+
+  if (!frequency.throttle) {
+    return null;
+  }
+
+  const value = frequency.throttle.slice(0, frequency.throttle.length - 1);
+  const unit = frequency.throttle.charAt(frequency.throttle.length - 1);
```

Comment:
See if you can reuse `getTimeTypeValue` from `x-pack/plugins/security_solution/public/detection_engine/rule_creation_ui/pages/rule_creation/helpers.ts`. 
That function needs a little refactoring, but it provides some error checking, has tests and I think is easier to understand that we are extracting time values and units than these two lines.
One thing I already see: for some reason it only accepts `['s', 'm', 'h']` time units, so you'll have to change that.

Reply:
Here's a small refactoring I attempted, with some comments from Maxim to solve: https://github.com/elastic/kibana/pull/157144/files#diff-3df6b92195fcf1a3174ce2f83d6c7ab11bbaf6bace416f776cede92746b4c5b8R32